activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ExpressionResolver.java

```diff
@@ -114,9 +114,8 @@ private Object resolveObjectPlaceHolder(ExpressionEvaluator expressionEvaluator,
             return expressionEvaluator.evaluate(expressionManager.createExpression(sourceString), expressionManager,
                 delegateInterceptor);
         } catch (final Exception e) {
-            logger.warn("Unable to resolve expression in variables, keeping original value",
-                        e);
-            return sourceString;
+            logger.warn("Unable to resolve expression in variables", e);
+            return null;
```

Comment:
we should never return null

Reply:
Not sure what else to return since the method returns an object that can't be resolved. `null` will be assigned to the variable with expression value that can't be resolved.